app/src/main/java/fi/kroon/vadret/presentation/AlertFragment.kt

```diff
@@ -0,0 +1,96 @@
+package fi.kroon.vadret.presentation
+
+import android.os.Bundle
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.snackbar.Snackbar
+import fi.kroon.vadret.R
+import fi.kroon.vadret.data.alert.exception.AlertFailure
+import fi.kroon.vadret.data.alert.model.Alert
+import fi.kroon.vadret.data.alert.model.Warning
+import fi.kroon.vadret.data.exception.Either
+import fi.kroon.vadret.data.exception.Failure
+import fi.kroon.vadret.presentation.adapter.AlertAdapter
+import fi.kroon.vadret.presentation.viewmodel.AlertViewModel
+import fi.kroon.vadret.utils.Schedulers
+import fi.kroon.vadret.utils.extensions.toVisible
+import fi.kroon.vadret.utils.extensions.viewModel
+import io.reactivex.rxkotlin.addTo
+import kotlinx.android.synthetic.main.alert_fragment.*
+import timber.log.Timber
+import javax.inject.Inject
+
+class AlertFragment : BaseFragment() {
+
+    override fun layoutId() = R.layout.alert_fragment
+
+    @Inject
+    lateinit var schedulers: Schedulers
+
+    @Inject
+    lateinit var alertAdapter: AlertAdapter
+
+    private lateinit var alertViewModel: AlertViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        cmp.inject(this)
+        alertViewModel = viewModel(viewModelFactory) {}
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initialiseView()
+        loadAlerts()
+    }
+
+    private fun initialiseView() {
+        alertRv.layoutManager = LinearLayoutManager(this.context, RecyclerView.VERTICAL, false)
+        alertRv.adapter = alertAdapter
+        alertRv.hasFixedSize()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        alertRv.adapter = null
+    }
+
+    private fun loadAlerts() {
+        alertRv.toVisible()
+        alertViewModel
```

Comment:
According to refactor todo list, we should move rxjava observables to viewmodel and deeper. Also, we should move `subscribeOn() / observeOn()` to use case